### PR TITLE
Additive nushellacturing.

### LINKED ISSUE #258: Inc with flags seems to only work if you pass a field

### PR DESCRIPTION
**Pull request by androbtech**
Began restructuring tests. Covered inc plugin cases and fixes #258 which applies for both the str & inc plugins.

### CODE REVIEW

File: src/plugins/inc.rs [hunk 1]
```diff
@@ -92,21 +124,21 @@ impl Plugin for Inc {
 
         Ok(Signature {
             name: "inc".to_string(),
-            positional: vec![PositionalType::optional_any("Field")],
+            positional: vec![],
```
Comment (jonathandturner) [inline src/plugins/inc.rs]: Curious - why remove the optional positional type?

File: src/plugins/str.rs [hunk 2]
```diff
@@ -129,7 +125,7 @@ impl Plugin for Str {
 
         Ok(Signature {
             name: "str".to_string(),
-            positional: vec![PositionalType::optional_any("Field")],
+            positional: vec![],
```
Comment (jonathandturner) [inline src/plugins/str.rs]: Same question here

File: src/plugins/str.rs [hunk 3]
```diff
@@ -224,11 +218,11 @@ mod tests {
             self
         }
 
-        fn create(&self, name_span: Span) -> CallInfo {
+        fn create(&self) -> CallInfo {
             CallInfo {
                 args: EvaluatedArgs::new(Some(self.positionals.clone()), Some(self.flags.clone())),
                 source_map: SourceMap::new(),
-                name_span,
+                name_span: Span::unknown(),
```
Comment (jonathandturner) [inline src/plugins/str.rs]: We're going from having span information to not having any?

File: src/plugins/inc.rs (hunk 1 above)
Comment (androbtech) [reply src/plugins/inc.rs]: Temporal workaround. You get a parse error when trying #258 and the Signature is configured like that. If we pass in an empty vec it works.

**Pull request merged**